storage/mysql/log_storage_test.go

```diff
@@ -153,44 +144,34 @@ func TestBegin(t *testing.T) {
 		if tx.(*logTreeTX).allowDuplicates != test.allowDuplicates {
 			t.Errorf("tx.allowDuplicates = %v, want = %v", tx.(*logTreeTX).allowDuplicates, test.allowDuplicates)
 		}
-
 		root, err := tx.LatestSignedLogRoot()
 		if err != nil {
 			t.Errorf("LatestSignedLogRoot() = (_, %v), want = (_, nil)", err)
 		}
-
 		if got, want := tx.WriteRevision(), root.TreeRevision+1; got != want {
 			t.Errorf("WriteRevision() = %v, want = %v", got, want)
 		}
-
-		if err := tx.Commit(); err != nil {
-			t.Errorf("Commit() = %v, want = nil", err)
-		}
-
+		commit(tx, t)
 	}
 }
 
 func TestSnapshot(t *testing.T) {
-	logID := createLogID("TestSnapshot")
 	cleanTestDB(DB)
-	prepareTestLogDB(DB, logID, t)
-
-	storage, err := NewLogStorage(DB)
-	if err != nil {
-		t.Fatalf("NewLogStorage() = (_, %v)", err)
-	}
+	logID := createLogForTests(DB)
+	s := NewLogStorage(DB)
 
 	tests := []struct {
 		logID int64
 		err   string
 	}{
 		{logID: -1, err: "failed to get tree row"},
-		{logID: logID.logID},
+		{logID: logID},
 	}
 
-	ctx := context.TODO()
 	for _, test := range tests {
-		tx, err := storage.SnapshotForTree(ctx, test.logID)
+		ctx := context.Background()
```

Comment:
My thought was that each test should have a separate context. I'm not entirely sure how long a context should live, TBH.

(Pushing back for now, please reopen if you think it's best.)